src/screens/MainQuizScreen.tsx

```diff
@@ -91,12 +105,14 @@ function MainQuizScreen() {
           />
         </div>
       )}
-      {!quizEnded && (
-        <Button
-          label="End Quiz"
-          onClick={handleEndQuiz}
-          className="end-quiz-button"
-        />
+      <Button
+        label="End Quiz"
+        onClick={handleEndQuiz}
+        className="end-quiz-button"
+      />
+
+      {showEndQuizModal && (
```

Comment:
I recommend you to remove it, because very soon we'll work with animation and this condition will break it
let's move `show` state inside the modal

Reply:
Can you give me some advice on how to do it better, please? Is useRef needed in EndQuizModal component?